docs/guides/server/hostname2.adoc

```diff
@@ -0,0 +1,161 @@
+* Frontend
+* Backend
+* Administration
+
+If you want to work with either of these endpoints, you need to set the base URL. The base URL consists of a several parts:
+
+* a scheme (e.g. https protocol)
+* a hostname (e.g. example.keycloak.org)
+* a port (e.g. 8443)
+* a path (e.g. /auth)
+
+The base URL for each group has an important impact on how tokens are issued and validated, on how links are created for actions that require the user to be redirected to Keycloak (for example, when resetting password through email links), and, most importantly, how applications will discover these endpoints when fetching the OpenID Connect Discovery Document from `realms/{realm-name}/.well-known/openid-configuration`.
+
+=== Frontend
+
+Users and applications use the frontend URL to access Keycloak through a front channel. The front channel is a communication channel, which is publicly accessible. For example browser-based flows (accessing the login page, clicking on the link to reset a password or binding the tokens).
+
+=== Backend
+
+The backend endpoints are those accessible through a public domain or through a private network. They're related to direct backend communication between Keycloak and a client (an application secured by Keycloak). Such communication might be over a local network, avoiding a reverse proxy. Examples of the endpoints that belong to this group are authorization endpoint, token and token introspection endpoint, userinfo endpoint, JWKS URI endpoint, etc.
+
+
+=== Administration
+
+Similarly to the base frontend URL, you can also set the base URL for resources and endpoints available through the management interface such as the administration console. The server exposes the administration console and static resources using a specific URL. This URL is used for redirect URLs, loading resources (CSS, JS), etc.
+
+== Sources for resolving the URL
+
+As indicated in the previous sections, URLs can be resolved in several ways: they can be dynamically generated, hardcoded, or a combination of both:
+
+* Dynamic from an incoming request:
+** Host header, scheme, server port, context path
+** Proxy-set headers: `Forwarded` and `X-Forwarded-*`
+* Hardcoded:
+** Server-wide config (e.g `--hostname`, `--hostname-admin`, etc.)
+** Realm configuration for frontend URL
+
+== Validations
+
+* `--hostname` URL and `--hostname-admin` URL are verified that full URL is used, incl. scheme and hostname. Port is validated only if present, otherwise default port for given protocol is assumed (80 or 443).
+* In production profile (`kc.sh|bat start`), either `--hostname` or `--hostname-strict` false must be explicitly configured.
+** This does not apply for dev profile (`kc.sh|bat start-dev`) where `--hostname-strict false` is the default value.
+* If `--hostname` is not configured:
+** `--hostname-backchannel-dynamic` must be set to false.
+** `--hostname-strict` must be set to false.
+* If `--hostname` is configured, --hostname-strict must be set to true.
```

Comment:
This validation was in the end removed from the implementation:
https://github.com/vmuzikar/keycloak/blob/c34e8d06d4078725f402037652f32874c55842a0/services/src/main/java/org/keycloak/url/HostnameV2ProviderFactory.java#L53-L56

Reply:
Addressed.